plugins/home/src/homePageComponents/StarredEntities/Content.tsx

```diff
@@ -55,25 +51,17 @@ export const Content = ({
       return [];
     }
 
-    const filter = [...starredEntities]
-      .map(ent => parseEntityRef(ent))
-      .map(ref => ({
-        kind: ref.kind,
-        'metadata.namespace': ref.namespace,
-        'metadata.name': ref.name,
-      }));
-
     return (
-      await catalogApi.getEntities({
-        filter,
+      await catalogApi.getEntitiesByRefs({
+        entityRefs: [...starredEntities],
         fields: [
           'kind',
           'metadata.namespace',
           'metadata.name',
           'metadata.title',
         ],
       })
-    ).items;
+    ).items.filter((e): e is Entity => !!e);
```

Comment:
To filter out `undefined` and make TypeScript compiler happy. Do you have a standard way of doing this in Backstage?

Reply:
This is fine or you could do

```typescript
.filter((e): e is Entity => Boolean(e));
```